### PR TITLE
FAQ: add FAQ questions and update format

### PR DESCRIPTION
**Pull request by lhyPingcap**
update FAQ
@lilin90 @zhengwanbo @sunhao2017 @dbaoutdo PTAL

### CODE REVIEW

**Comment by meyu44** [comment]
@shenli PTAL

File: FAQ.md [hunk 1]
```diff
@@ -203,6 +254,35 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 
 使用 docker-compose 在本地一键拉起一个集群，包括集群监控，还可以根据需求自定义各个组件的软件版本和实例个数，以及自定义配置文件，这种只限于开发环境，详细可参考：[官方文档](https://pingcap.com/docs-cn/op-guide/docker-compose/)。
 
+#### 2.2.4 如何开启 TiDB 中的慢查询日志？
```
Comment (shenli) [inline FAQ.md]: 这个有歧义，简单来说，tidb的慢查询日志一直有，只是打印在哪个文件里的问题，另外可以写一下如何在日志中找到慢查询相关信息

File: FAQ.md [hunk 2]
```diff
@@ -773,3 +895,7 @@ update mysql.tidb set variable_value='30m' where variable_name='tikv_gc_life_tim
 #### 9.2.3 ERROR 1148 (42000): the used command is not allowed with this TiDB version 问题的处理方法？
 
 这个问题是因为在执行 `LOAD DATA LOCAL` 语句的时候，MySQL 客户端不允许执行此语句（即 `local_infile` 选项为 0）。解决方法是在启动 MySQL 客户端时，用 `--local-infile=1` 选项。具体启动指令类似：`mysql --local-infile=1 -u root -h 127.0.0.1 -P 4000`。有些 MySQL 客户端需要设置而有些不需要设置，原因是不同版本的 MySQL 客户端对 `local-infile` 的默认值不同。
+
+#### 9.2.4 ERROR 9001 (HY000): PD server timeoutstart timestamp may fall behind safepoint
+
+这个报错一般是 TiDB 访问 PD 出了问题，TiDB 后台有个 worker 会不断地从 PD 查询 safepoint，如果超过 100s 查不成功就会报这个错。TiDB 常见错误码请参考 [错误码与故障诊断](https://pingcap.com/docs-cn/sql/error/)。
```
Comment (shenli) [inline FAQ.md]: 这里加一下如何诊断以及解决

File: FAQ.md [hunk 3]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
 
 一个事务中的语句数量，默认限制最大为 5000 条。
 
+#### 1.1.19 TiDB 是否支持 XA？
+
+虽然 TiDB 的 JDBC 驱动用的就是 MySQL JDBC（Connector / J）, 但是当使用 Atomikos 的时候，数据源要配置成类似这样的配置：`type="com.mysql.jdbc.jdbc2.optional.MysqlXADataSource"`。MySQL JDBC XADataSource 连接 TiDB 的模式目前是不支持的。MySQL JDBC 中配置好的 XADataSource 模式，只对 MySQL 数据库起作用（dml 去修改 redo 等）。
```
Comment (lilin90) [inline FAQ.md]: - Please pay attention to the punctuation: change the first comma to the Chinese version.
- dml -> DML

File: FAQ.md [hunk 4]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+Atomikos 配好两个数据源后，JDBC 驱动都要设置成 XA 模式，然后 Atomikos 在操作 TM 和 RM（DB）的时候，会通过数据源的配置，发起带有 XA 指令到 JDBC 层，JDBC 层 XA 模式启用的情况下，会对 InnoDB（如果是 MySQL 的话）下发操作一连串 XA 逻辑的动作，包括 DML 去变更 redo log 等，就是两阶段递交的那些操作。TiDB 的引擎目前版本中，没有对上层应用层 JTA / XA 的支持，是不解析这些 Atomikos 发过来的 XA 类型的操作的。
```
Comment (lilin90) [inline FAQ.md]: Change "TiDB 的引擎目前版本中，没有对上层应用层 JTA / XA 的支持，是不解析这些 Atomikos 发过来的 XA 类型的操作的。" to:

"TiDB 目前的引擎版本中，没有对上层应用层 JTA / XA 的支持，不解析这些 Atomikos 发过来的 XA 类型的操作。"

File: FAQ.md [hunk 5]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+虽然 TiDB 的 JDBC 驱动用的就是 MySQL JDBC（Connector / J）, 但是当使用 Atomikos 的时候，数据源要配置成类似这样的配置：`type="com.mysql.jdbc.jdbc2.optional.MysqlXADataSource"`。MySQL JDBC XADataSource 连接 TiDB 的模式目前是不支持的。MySQL JDBC 中配置好的 XADataSource 模式，只对 MySQL 数据库起作用（dml 去修改 redo 等）。
+
+Atomikos 配好两个数据源后，JDBC 驱动都要设置成 XA 模式，然后 Atomikos 在操作 TM 和 RM（DB）的时候，会通过数据源的配置，发起带有 XA 指令到 JDBC 层，JDBC 层 XA 模式启用的情况下，会对 InnoDB（如果是 MySQL 的话）下发操作一连串 XA 逻辑的动作，包括 DML 去变更 redo log 等，就是两阶段递交的那些操作。TiDB 的引擎目前版本中，没有对上层应用层 JTA / XA 的支持，是不解析这些 Atomikos 发过来的 XA 类型的操作的。 
+
+MySQL 是单机数据库，只能通过 XA 来满足跨数据库事务，而 TiDB 本身就通过 Google 的 Percolator 事务模型支持分布式事务，性能稳定性比 XA 要高出很多，所以不会也不需要支持 XA。
+
+#### 1.1.20 show processlist 是否显示系统线程号？
+
+TiDB 的 `show processlist` 与 mysql 的 `show processlist` 显示内容基本一样，不会显示系统进程号，而 ID 表示当前的 session ID。其中 TiDB 的 `show processlist` 和 mysql 的 `show processlist` 区别如下：
```
Comment (lilin90) [inline FAQ.md]: mysql -> MySQL

**Please correct all "mysql"s in this PR to "MySQL".**

File: FAQ.md [hunk 6]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+1）由于 TiDB 是分布式数据库，tidb-server 实例是无状态的 SQL 解析和执行引擎（详情可参考 [TiDB 整体架构](https://pingcap.com/docs-cn/overview/#tidb-整体架构)），用户使用 mysql 客户端登录的是哪个 tidb-server，`show processlist` 就会显示当前连接的这个 tidb-server 中执行的 session 列表，不是整个集群中运行的全部  session 列表；而 mysql 是单机数据库，`show processlist` 列出的是当前整个 mysql 数据库的全部执行 SQL 列表。
```
Comment (lilin90) [inline FAQ.md]: - Delete the extra whitespace after "全部".
- **Change all the links involved in this PR to the relative path link.** For example: `[TiDB 整体架构](overview.md#tidb-整体架构)`

File: FAQ.md [hunk 7]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+2）TiDB 的 `show processlist` 显示内容比起 mysql 来讲，多了一个当前 session 使用内存的估算值（单位 Byte）。
+
+#### 1.1.21 如何修改用户名密码和权限？
+
+TiDB 作为分布式数据库，在 TiDB 中修改用户密码建议使用 `set password for 'root'@'%' = '0101001';` 或 `alter` 方法，不推荐使用 `update mysql.user` 的方法进行，这种方法可能会造成其它节点刷新不及时的情况。修改权限也一样，都建议采用官方的标准语法。详情可参考 [TiDB 用户账户管理](https://pingcap.com/docs-cn/sql/user-account-management/)。
```
Comment (lilin90) [inline FAQ.md]: Ditto. relative path link

File: FAQ.md [hunk 8]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+#### 1.1.22 TiDB中，为什么出现后插入数据的自增 ID 反而小？
+
+TiDB 的自增 ID (AUTO\_INCREMENT ID) 只保证自增且唯一，并不保证连续分配。TiDB 目前采用批量分配的方式，所以如果在多台 TiDB 上同时插入数据，分配的自增 ID 会不连续。当多个线程并发往不同的 tidb-server 插入数据的时候，有可能会出现后插入的数据自增 ID 小的情况。此外，TiDB允许给整型类型的字段指定 AUTO_INCREMENT，且一个表只允许一个属性为 AUTO_INCREMENT 的字段。详情可参考 [数据定义语言](https://pingcap.com/docs-cn/sql/ddl/)。
```
Comment (lilin90) [inline FAQ.md]: - Add inline code format `` to `AUTO_INCREMENT`.
- Ditto. relative path link

File: FAQ.md [hunk 9]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+#### 1.1.23 sql_mode 默认除了通过命令 set 修改，配置文件怎么修改？
+
+TiDB 的 sql_mode 与 mysql 的 sql_mode 设置方法有一些差别，TiDB 不支持配置文件配置设置数据库的 sql\_mode，而只能使用 set 命令去设置，具体方法为：`set @@global.sql_mode = 'STRICT_TRANS_TABLES';`。
+
+#### 1.1.24 我们支持哪些认证协议，过程是怎样的？
```
Comment (lilin90) [inline FAQ.md]: "我们" -> "TiDB"

File: FAQ.md [hunk 10]
```diff
@@ -84,6 +84,47 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
...
+
+这一层跟 MySQL 一样，走的 SASL 认证协议，用于用户登陆认证，对密码的处理流程。
+
+客户端连接 TiDB 的时候，走的是 challenge / response（挑战-应答）的认证模式，过程如下：
+第一步：客户端连接服务器；
```
Comment (lilin90) [inline FAQ.md]: Add a blank line before "第一步"

File: FAQ.md [hunk 11]
```diff
@@ -118,7 +159,7 @@ TiDB 字符集默认就是 UTF8 而且目前只支持 UTF8，字符串就是 mem
 
 ##### 2.1.1.1  为什么要在 CentOS 7 上部署 TiDB 集群？
 
-TiDB 作为一款开源分布式 NewSQL 数据库，可以很好的部署和运行在 Intel 架构服务器环境及主流虚拟化环境，并支持绝大多数的主流硬件网络，作为一款高性能数据库系统，TiDB 支持主流的 Linux 操作系统环境，具体可以参考 TiDB 的[官方部署要求](https://pingcap.com/docs-cn/op-guide/recommendation/)。 其中 TiDB 在 CentOS 7.3 的环境下进行大量的测试，同时也有很多这个操作系统的部署最佳实践，因此，我们推荐客户在部署 TiDB 的时候使用 CentOS 7.3+ 以上的Linux 操作系统。
+TiDB 作为一款开源分布式 NewSQL 数据库，可以很好的部署和运行在 Intel 架构服务器环境及主流虚拟化环境，并支持绝大多数的主流硬件网络，作为一款高性能数据库系统，TiDB 支持主流的 Linux 操作系统环境，具体可以参考 TiDB 的 [官方部署要求](https://pingcap.com/docs-cn/op-guide/recommendation/)。 其中 TiDB 在 CentOS 7.3 的环境下进行大量的测试，同时也有很多这个操作系统的部署最佳实践，因此，我们推荐客户在部署 TiDB 的时候使用 CentOS 7.3+ 以上的Linux 操作系统。
```
Comment (lilin90) [inline FAQ.md]: - Delete the unnecessary whitespace before "[官方部署]"
- link: relative path

File: FAQ.md [hunk 12]
```diff
@@ -151,6 +192,16 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
 
 资源可接受的话，我们建议做 RAID 10，如果资源有限，也可以不做 RAID。
 
+##### 2.1.2.5 TiDB 集群各个组件的配置推荐？
+
+tidb-server 需要 CPU 和内存比较好的机器，参考官网配置要求，如果后期需要开启 binlog，根据业务量的评估和 GC 时间的要求，也需要本地磁盘大一点，不要求 SSD 磁盘；
+
+pd-server 里面存了集群元信息，会有频繁的读写请求，对磁盘 IO 要求相对比较高，磁盘太差会影响整个集群性能，推荐 SSD 磁盘，空间不用太大。另外集群 region 数量越多对 CPU、内存的要求越高；
+
+tikv-server 对 CPU、内存、磁盘要求都比较高，一定要用 SSD 磁盘。
```
Comment (lilin90) [inline FAQ.md]: For the above three items, using a bulleted list would make them clearer.

File: FAQ.md [hunk 13]
```diff
@@ -151,6 +192,16 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+
+详情可参考 [TiDB 软件和硬件环境要求](https://pingcap.com/docs-cn/op-guide/recommendation/)
```
Comment (lilin90) [inline FAQ.md]: Ditto. link

File: FAQ.md [hunk 14]
```diff
@@ -203,6 +254,35 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+如果希望生成单独的慢查询日志文件，修改 inventory.ini 配置文件的参数 `enable_slow_query_log` 为 True。
+
+如上配置修改之后，需要执行 `ansible-playbook rolling_update.yml --tags=tidb`，对 tidb-server 实例进行滚动升级，升级完成后，tidb-server 将在 `tidb_slow_query.log`
+文件中记录慢查询日志。
+
+#### 2.2.5 首次部署 TiDB 集群时，没有配置 tikv 的 Label 信息，在后续如何添加配置 Label？
+
+TiDB 的 Label 设置是与集群的部署架构相关的，是集群部署中的重要内容，是 PD 进行全局管理和调度的依据。如果集群在初期部署过程中没有设置 Label，需要在后期对部署结构进行调整，就需要手动通过 PD 的管理工具 pd-ctl 来添加 location-labels 信息，例如：`config set location-labels "zone, rack, host"`（根据实际的 label 层级名字配置）。
+
+pd-ctl 的使用参考 [PD Control 使用说明](https://pingcap.com/docs-cn/tools/pd-control/)
```
Comment (lilin90) [inline FAQ.md]: Ditto. link

File: FAQ.md [hunk 15]
```diff
@@ -203,6 +254,35 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+
+#### 2.2.6 为什么测试磁盘的 dd 命令用 oflag=direct 这个选项？
+
+Direct 模式就是把写入请求直接封装成 IO 指令发到磁盘，这样是为了绕开文件系统的缓存，可以直接测试磁盘的真实的 IO 读写能力。
```
Comment (lilin90) [inline FAQ.md]: IO -> I/O

File: FAQ.md [hunk 16]
```diff
@@ -203,6 +254,35 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+
+#### 2.2.7 如何用 fio 命令测试 TiKV 实例的磁盘性能
```
Comment (lilin90) [inline FAQ.md]: Add a question mark in the end

File: FAQ.md [hunk 17]
```diff
@@ -203,6 +254,35 @@ TiDB 支持部署和运行在 Intel x86-64 架构的 64 位通用硬件服务器
...
+
+随机读测试：
+
+`./fio -ioengine=libaio -bs=32k -direct=1 -thread -rw=randread  -size=10G -filename=fio_randread_test.txt -name='PingCAP' -iodepth=4 -runtime=60`
```
Comment (lilin90) [inline FAQ.md]: Change the inline code format to code block format ```.

File: FAQ.md [hunk 18]
```diff
@@ -266,7 +346,7 @@ Binary 不是我们建议的安装方式，对升级支持也不友好，建议
 
 #### 3.1.5 TiDB 有哪些系统表？
 
-和 MySQL 类似，TiDB 中也有系统表，用于存放数据库运行时所需信息，具体信息参考：[TiDB 系统数据库](https://pingcap.com/docs-cn/sql/system-database/)文档。
+和 MySQL 类似，TiDB 中也有系统表，用于存放数据库运行时所需信息，具体信息参考：[TiDB 系统数据库](https://pingcap.com/docs-cn/sql/system-database/) 文档。
```
Comment (lilin90) [inline FAQ.md]: - link
- Delete the extra whitespace

File: FAQ.md [hunk 19]
```diff
@@ -382,7 +462,7 @@ Client 连接只能通过 TiDB 访问集群，TiDB 负责连接 PD 与 TiKV，PD
 
 #### 3.3.4 Infomation_schema 能否支持更多真实信息？
 
-Infomation_schema 库里面的表主要是为了兼容 MySQL 而存在，有些第三方软件会查询里面的信息。在目前 TiDB 的实现中，里面大部分只是一些空表。后续随着 TiDB 的升级，会提供更多的参数信息。当前 TiDB 支持的：Infomation\_schema 请参考[TiDB 系统数据库说明文档](https://pingcap.com/docs-cn/sql/system-database)。
+Infomation_schema 库里面的表主要是为了兼容 MySQL 而存在，有些第三方软件会查询里面的信息。在目前 TiDB 的实现中，里面大部分只是一些空表。后续随着 TiDB 的升级，会提供更多的参数信息。当前 TiDB 支持的：Infomation\_schema 请参考 [TiDB 系统数据库说明文档](https://pingcap.com/docs-cn/sql/system-database)。
```
Comment (lilin90) [inline FAQ.md]: - link
- Delete the extra whitespace after "参考"

File: FAQ.md [hunk 20]
```diff
@@ -412,7 +496,7 @@ TiKV 本地存储的 cluster ID 和指定的 PD 的 cluster ID 不一致。在
 
 #### 3.4.3 TiKV 启动报错：duplicated store address
 
-启动参数中的地址已经被其他的 TiKV 注册在 PD 集群中了。造成该错误的常见情况：TiKV `--data-dir` 指定的路径下没有数据文件夹（删除或移动后没有更新 --data-dir），用之前参数重新启动该 TiKV。请尝试用 pd-ctl 的[store delete](https://github.com/pingcap/pd/tree/master/pdctl#store-delete-store_id)功能，删除之前的 store, 然后重新启动 TiKV 即可。
+启动参数中的地址已经被其他的 TiKV 注册在 PD 集群中了。造成该错误的常见情况：TiKV `--data-dir` 指定的路径下没有数据文件夹（删除或移动后没有更新 --data-dir），用之前参数重新启动该 TiKV。请尝试用 pd-ctl 的 [store delete](https://github.com/pingcap/pd/tree/master/pdctl#store-delete-store_id) 功能，删除之前的 store, 然后重新启动 TiKV 即可。
```
Comment (lilin90) [inline FAQ.md]: Use the Chinese version for the last comma.

File: FAQ.md [hunk 21]
```diff
@@ -578,6 +666,14 @@ DB2、Oracle 到 TiDB 数据迁移（增量+全量），通常做法有：
 
 没有，目前依赖程序自行实现。
 
+##### 4.2.1.4 利用 Syncer 做数据同步的时候是否支持只同步部分表？
+
+支持，具体参考 syncer 使用手册 [Syncer 使用文档](https://pingcap.com/docs-cn/tools/syncer/)
```
Comment (lilin90) [inline FAQ.md]: - syncer -> Syncer
- link

File: FAQ.md [hunk 22]
```diff
@@ -599,7 +695,7 @@ TiDB 读流量可以通过增加 TiDB server 进行扩展，总读容量无限
 - KV entry 的总条数不超过 30w
 - KV entry 的总大小不超过 100MB
 
-在 Google 的 Cloud Spanner 上面，也有类似的[限制](https://cloud.google.com/spanner/docs/limits)。
+在 Google 的 Cloud Spanner 上面，也有类似的 [限制](https://cloud.google.com/spanner/docs/limits)。
```
Comment (lilin90) [inline FAQ.md]: **Please delete all the similar whitespace before linked Chinese words involved in this PR.**

File: FAQ.md [hunk 23]
```diff
@@ -58,15 +58,15 @@ TiDB 支持 ACID 分布式事务，事务模型是以 Google 的 Percolator 模
...
 
-目前 [官方文档](https://pingcap.com/docs-cn/)是获取 TiDB 相关知识最主要、最及时的发布途径。除此之外，我们也有一些技术沟通群，如有需求可发邮件至 [info@pingcap.com](mailto:info@pingcap.com) 获取。
+目前[官方文档](overview.md#tidb-简介)是获取 TiDB 相关知识最主要、最及时的发布途径。除此之外，我们也有一些技术沟通群，如有需求可发邮件至 [info@pingcap.com](mailto:info@pingcap.com) 获取。
 
 #### 1.1.14 TiDB 对那些 MySQL variables 兼容？
 
-详细可参考：[系统变量](sql/variable.md)
+详细可参考[系统变量](sql/variable.md)。
 
 #### 1.1.15 TiDB 是否支持 select for update 吗？
```
Comment (lilin90) [inline FAQ.md]: -> TiDB 是否支持 select for update？

**Comment by zhengwanbo** [comment]
LGTM
